k8s-health-agent/files/k8s-health-agent-deployment.yaml

```diff
@@ -0,0 +1,65 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app.kubernetes.io/managed-by: kustomize
+    app.kubernetes.io/name: k8s-health-agent
+    control-plane: controller-manager
+  name: k8s-health-agent-controller-manager
+  namespace: k8s-health-agent-system
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      control-plane: controller-manager
```

Comment:
Nit pick: I'm slightly worried about this label. Seems a bit too wide. What happens when we want to deploy more controllers into the destination namespace in the future?